lib/puppet/util/http_proxy.rb

```diff
@@ -127,6 +127,20 @@ def self.http_proxy_password
     return Puppet.settings[:http_proxy_password]
   end
 
+  def self.no_proxy
+    no_proxy_env = ENV["no_proxy"] || ENV["NO_PROXY"]
+
+    if no_proxy_env
+      return no_proxy_env
+    end
+
+    if Puppet.settings[:no_proxy] == 'none'
+      return nil
+    end
+
+    return Puppet.settings[:no_proxy]
```

Comment:
Can we just make the default `nil` so we don't need to check for `none` above?

Reply:
I don't think so. Checking for 'none' is modeled after the code for `http_proxy_host` and `http_proxy_password`. It seems that using 'none' is an option when the setting these via ENV variables. I double-checked this when I first worked on the code, and after confirming it now (search for `"http_proxy=none"` it seems that 'None' is also valid, so, if anything we could perform a case-insensitive compare ... though no one has raised an issue regarding that.